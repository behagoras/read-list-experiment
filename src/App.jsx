import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
} from 'react-router-dom';

import Layout from './layouts/AppLayout';

// Routes
import Home from './pages/Home';

// Redux
import mockStore from './mocks/store.json';
import reducers from './reducers';
import GlobalStyles from './styles/globalStyles';

// Initialize store
const store = createStore(reducers, { original: mockStore, filtered: mockStore });;

const App = (props) => {
  return (
    <Provider store={store}>
      <Router>
        <GlobalStyles />
        <Switch>
          <Route path="/">
            <Layout>
              <Home />
            </Layout>
          </Route>
        </Switch>
      </Router>
    </Provider>
  );
};

export default App;
