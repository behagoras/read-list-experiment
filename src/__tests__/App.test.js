import React from 'react';
import 'jest-styled-components';
import toJson from 'enzyme-to-json';

import { shallow } from 'enzyme';

import App from '../App';

describe('<App />', () => {
  const output = shallow(<App />);
  const jsonOutput = toJson(output);

  it('Should match Snapshot and render an app', () => {
    expect(jsonOutput).toMatchSnapshot();
  });
  it('Should have a BrowserRouter as the first child', () => {
    expect(toJson(output).children[0].type).toBe('BrowserRouter');
  });
});
