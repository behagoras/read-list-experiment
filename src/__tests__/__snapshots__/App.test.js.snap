// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should match Snapshot and render an app 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <div>
      <nav>
        <ul>
          <li>
            <Link
              to="/"
            >
              Homes
            </Link>
          </li>
        </ul>
      </nav>
      <Switch>
        <Route
          path="/"
        >
          <Connect(AppLayout)>
            <Connect(Home) />
          </Connect(AppLayout)>
        </Route>
      </Switch>
    </div>
  </BrowserRouter>
</Provider>
`;
